### ISSUE TITLE
Disappears after setting dock to the bottom in some configurations

### ISSUE THREAD

**Issue by jacobgkau** [pop-os/cosmic-workspaces#44]
Discovered while testing https://github.com/pop-os/cosmic-workspaces/pull/41 (but not related to that PR):

**Steps to recreate:**
1. Set the settings to:
        - Dock always hide
        - Dock extends to edges
        - Dock on the right
        - Workspace picker on the right
2. Restart GNOME Shell
3. Move dock to bottom

**Actual behavior:** Workspace picker disappears until next GNOME Shell restart, or until dock is moved to either side again.

**Expected behavior:** Picker doesn't disappear.

**Other notes:** Screenshot of the issue in https://github.com/pop-os/cosmic-workspaces/pull/41#issuecomment-1024691763.

**Comment by 13r0ck** [comment]
What display did you see this on? I am not immediately seeing this

**Comment by 13r0ck** [comment]
I saw this too. I'm not sure exactly what I did to make it happen either...
I didn't have to restart gnome shell though

**Comment by XV-02** [comment]
I think I have a method for reproduction:

### Steps

1. With dynamic workspaces, have two workspaces open, one active and one standby
2. Enable dock
3. Set visibility to "Always hide" or "Intelligently hide"
4. Set dock location to either left or right side of screen
5. Enter Workspace View
6. Leave Workspace view
7. Change dock location to bottom of screen
8. Enter Workspace View

At this point you should see something similar to the following:

![Screenshot from 2022-02-02 15-58-57](https://user-images.githubusercontent.com/98765732/152251894-adc174fb-7000-47aa-8045-509f67f30ef2.png)

This screenshot is with the Workspaces view open.

**Summary:**
With the cosmic dock enabled, transitioning from the dock from either the left or right of the screen to the bottom, while _also_ having either "Always hide" or "Intelligently hide" enabled, results in the Workspaces view failing to display the Workspaces picker.

### Notes
- This method was achieved on a Pangolin 10 using only the built in display, and hasn't yet been tested in multi-monitor set ups. _Further confirmed on a Gazelle 15 with built-in display and a Thelio Major R2 with single-monitor._
- There was no difference with respect to whether the dock extended to screen edges or not.
- The issue persisted to effect the Workspaces picker after disabling the dock in settings.
- The issue seemed agnostic to whether the picker was on the left or right hand side of the screen.
- The issue seemed agnostic as to whether the left or right side of the screen was set for the dock before transitioning to the bottom.
- The issue doesn't seem to arise unless using dynamic workspaces. Setting fixed workspaces to only two did not seem to trigger the issue.

**Comment by XV-02** [comment]
Additionally, I also noticed another issue which seemed contingent upon this one, and might help diagnostically:

### Description

After having set-up the larger disappearing workspace picker issue, I was able to cause the workspace picker to disappear and reappear by closing (disappear) or opening (reappear) a window in the standby workspace (transitioning from 1 active/ 1 standby to 2 active/ 1 standby) with dynamic workspaces. This looked as follows:

![disappearing-workspaces-2022-02-02](https://user-images.githubusercontent.com/98765732/152252539-0f2affc9-fb44-4e5e-b4b3-9ffd776380c9.gif)

This was achieved by opening a terminal in the second workspace via **Super + t** , and closing it by clicking the "X" bubble for the window in workspace view. This was repeated several times. The same behaviour was observed when opening Files with **Super + f** or by interacting with the dock itself to launch a window of any description. Much like the larger issue with the workspaces picker not displaying, this issue persisted after disabling the dock, but did not manifest when the larger workspace picker issue was itself not present.

### Steps
1. Set Workspaces to dynamic.
2. Check that Workspace Picker appears as intended.
2. With Workspace View operating as expected, open windowed applications of any description in the first two workspaces, such that you have 2 active workspaces, and one stand-by like so: 

![Screenshot from 2022-02-02 13-58-52](https://user-images.githubusercontent.com/98765732/152236404-961082f2-2765-4f79-9b60-2974fcbc512c.png)

**Note**: the first two workspaces contain windows, the third is empty.

3. Set-up the Workspace View bug [thusly](https://github.com/pop-os/cosmic-workspaces/issues/44#issuecomment-1028289273)
4. Enter the second workspace and, from the workspaces view, close all windows.

At this point, the previously visible workspaces picker should disappear.

5. Open a window in the second workspace either through the dock, or by keyboard shortcut.

At this point, the picker should reappear.

6. Repeat steps 4 & 5 as many times as require to exhaust the novelty.

**Summary**
Having produced the workspace picker issue with dynamic workspaces, moving from two to three or more workspaces causes the picker to appear again, while reverting back to two workspace causes the picker to disappear again. 

###Notes

- This method was achieved on a Pangolin 10 and a Gazelle 15 using built-in displays and a Thelio Major R2 with single-monitor. It hasn't yet been tested in multi-monitor set ups. 
- The behaviour was observed both with and without directly interacting with the dock itself or the picker itself. 
- As with the larger dock/picker issue herein, this behaviour continued after disabling the dock unless the dock had been reset to the sides such that the picker displayed as expected.
- There seemed to be no difference in behaviour whether or not one moved between workspaces, so long as at the end of the process, there was one empty workspace and one active.

**Comment by 13r0ck** [comment]
Does this still occur in 22.04 after #56 ?

**Comment by 13r0ck** [comment]
@jacobgkau 
> Does this still occur in 22.04 after https://github.com/pop-os/cosmic-workspaces/pull/56 ?

**Comment by flawpiesammen** [comment]
I have an additional bug where I have my workspaces on the right and the dock on the left side of the screen. Workspaces seem to be extending past the edge of the screen (drawn too far to the right). If I hide the dock the appear properly aligned.

**Comment by 13r0ck** [comment]
@Jewseephish please do not post unrelated issues in another issue. If you have an issue, please open a new issue.

**Comment by 13r0ck** [comment]
I believe this has been fixed since the changes to both cosmic-dock & cosmic-workspaces in 22.04
